src/runtime_src/core/tools/common/Process.cpp

```diff
@@ -163,14 +163,20 @@ XBUtilities::runScript( const std::string & env,
   }
 
   is_done = true;
-  bool exitCode = (os_stdout.str().find("PASS") != std::string::npos) ? true : false;
-  run_test.get()->finish(exitCode, final_description);
+  bool passed = (os_stdout.str().find("PASS") != std::string::npos) ? true : false;
+  bool skipped = (os_stdout.str().find("EOPNOTSUPP") != std::string::npos) ? true : false;

```

Comment:
Did you look at the code below (part of the #else section) to see if this SKIPPING data is needed?

Yeah, depending on the OS Ubuntu or CentOS, the boost library may or may not be the version we need, hence this compiler directive.

Reply:
Yes this is needed as the code is failing on Amazon linux which doesn't have boost::process

Reply:
We can capture the exact exit_code in the below section and don't have to handcraft the return code by parsing the output string thanks to <boost/process>, this is how it works in my ubuntu 18.04 machine.

The behavior of runTestCase() should be the same across all the environments, should not be failed on one OS and skipped on another even though parsing the output string to decide the return code is not ideal.

I think since we can get the exit_code in the else section, so we don't have to handcraft the return there.